pkg/controller/aws/cache/replicationgroup.go

```diff
@@ -321,10 +308,15 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 
 	// The group is unnamed. Assume it has not been created in AWS.
 	if rd.Status.GroupName == "" {
-		return reconcile.Result{Requeue: client.Create(ctx, rd)}, errors.Wrapf(r.kube.Update(ctx, rd), "cannot update resource %s", req.NamespacedName)
+		requeue, authToken := client.Create(ctx, rd)
```

Comment:
We do, but we reflect that by setting the resource's status to failed in the Kubernetes API as part of the `client.Create()` call rather than by returning an error. There's no change to that logic here - we just take the return values separately rather than inline.